newrelic/resource_newrelic_alert_policy_channel.go

```diff
@@ -24,55 +25,70 @@ func resourceNewRelicAlertPolicyChannel() *schema.Resource {
 				ForceNew: true,
 			},
 			"channel_id": {
-				Type:     schema.TypeInt,
-				Required: true,
-				ForceNew: true,
+				Type:          schema.TypeInt,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"channel_ids"},
+				Deprecated:    "use `channel_ids` argument instead",
+			},
+			"channel_ids": {
+				Type:          schema.TypeList,
+				Optional:      true,
+				ForceNew:      true,
+				MinItems:      1,
+				ConflictsWith: []string{"channel_id"},
+				Elem: &schema.Schema{
+					Type: schema.TypeInt,
+				},
 			},
 		},
 	}
 }
 
 func resourceNewRelicAlertPolicyChannelCreate(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*ProviderConfig).NewClient
+	policyChannels, err := expandAlertPolicyChannels(d)
 
-	policyID := d.Get("policy_id").(int)
-	channelID := d.Get("channel_id").(int)
+	if err != nil {
+		return err
+	}
 
-	serializedID := serializeIDs([]int{policyID, channelID})
+	serializedID := serializeIDs(append(
+		[]int{policyChannels.ID},
+		policyChannels.ChannelIDs...,
```

Comment:
If we don't sort these, changing their order in HCL will result in drift.   Probably not a big deal, but something to consider?

Reply:
Even if we sort the IDs and save them in order within the state, TF will still detect a change if the user changes the order of the IDs in the HCL in a way that doesn't match the state, which then creates a new resource due to `ForceNew: true`

Reply:
But sorting does avoid some form of drift, but there's still a chance a human change will cause drift. We can add a note in the docs to say "Best practice: sort your channel IDs in ascending order within your HCL to avoid state drift"

Reply:
👍